### ISSUE TITLE
x/build: time out on LUCI Linux/PPC64x builder

### ISSUE THREAD

**Issue by cherrymui** [golang/go#65171]
```
#!watchflakes
post <- builder ~ `(gotip|go1\.\d\d)-linux-ppc64` && (`test timed out` || `SIGQUIT` || `context deadline exceeded` || `running too slowly` || status == "ABORT") && test != "TestServerReadAfterWriteHeader100Continue/h1"
```

**Comment by pmur** [comment]
I wonder if LUCI needs to set `GO_TEST_TIMEOUT_SCALE=2` for ppc64{,le}? I don't think I see that set in the logs. The POWER8 builders can be particularly slow. I think the VMs are also subject to sharing CPU time with whatever else is running at OSU.

**Comment by cherrymui** [comment]
Yeah, `GO_TEST_TIMEOUT_SCALE=2` doesn't seem to be set. The old ppc64le builders set `GO_TEST_TIMEOUT_SCALE=2`, so we probably should do that for LCUI builders as well.

cc @mknyszek

**Comment by mknyszek** [comment]
I'll send a CL for this today. Thanks.

**Comment by cherrymui** [comment]
Closing. Should be fixed by the CL above. (Apparently gopherbot doesn't do it be cause it is checked in to a branch.)

**Comment by pmur** [comment]
Is there more context to be found about what test is suspected of hanging, and what the timeout is?

I am rebuilding the container images to include lsof.

**Comment by pmur** [comment]
Hrm, I wonder if this is timing out more frequently because the jobs are not running on a tmpfs. The only tmpfs I have mounted to the containers is /workdir (as was used by the old CI), it doesn't look to be used.

Is there an option to move the work/caches to a tmpfs? My initial though is to mount /home/swarming as a tmpfs.

**Comment by dmitshur** [comment]
Reports above seem to be because didn't have the `GO_TEST_TIMEOUT_SCALE=2` env var fully propagated to the environment of one of the two builders. That should be fixed for future builds ([CL 563396](https://go.dev/cl/563396)). Closing again optimistically, watchflakes will reopen if there turns out to be more to do after that.

**Comment by pmur** [comment]
I've also changed the docker configuration to mount /home/swarming as a tmpfs.

**Comment by pmur** [comment]
The latest batch of failures is the same issue as #65725. I only adjusted the RAM limits on the ppc64le LUCI containers. Looking through the logs, the ppc64-power10 LUCI containers also need more RAM.

**Comment by pmur** [comment]
The net/http failures should be attributed to #67382, noteably have not been mentioned on the other issue.  Reclosing this issue.

**Comment by pmur** [comment]
I inspected [ppc64-power10 failure ](https://ci.chromium.org/b/8746487826012416609) directly before this message. It shows many failures, but not those which caused the timeout.  Inspecting the go testing log, I see the timing out tests mentioned by gopherbot. Is that expected behavior?

**Comment by timothy-king** [comment]
Another dup of https://github.com/golang/go/issues/70363#issuecomment-2481646584

**Comment by rsc** [comment]
These are #78576: 

| Comment | Date | Builder | watchflakes entry | What actually hung |
|---|---|---|---|---|
| [4022744022](https://github.com/golang/go/issues/65171#issuecomment-4022744022) | 2026-03-08 | ppc64le_power10 | `net/http/httputil.TestReverseProxyWebSocketHalfTCP/server_close_write` [SKIP] | httputil, 18m0s; goroutine at `<autogenerated>:1 +0x10` |
| [4026279237](https://github.com/golang/go/issues/65171#issuecomment-4026279237) | 2026-03-09 | ppc64_power10 | `net/http/httputil.TestReverseProxy_Post` [SKIP] | httputil, 18m0s |
| [4156944468](https://github.com/golang/go/issues/65171#issuecomment-4156944468) | 2026-03-30 | ppc64le_power8 | `cmd/go.TestScript/mod_skip_write` | child `go mod why -m rsc.io/sampler` |
| [4356405850](https://github.com/golang/go/issues/65171#issuecomment-4356405850) | 2026-04-24 | ppc64le_power10 | `cmd/go.TestScript/mod_skip_write` | same |
| [4390440237](https://github.com/golang/go/issues/65171#issuecomment-4390440237) | 2026-05-06 | ppc64_power10 | `image/gif.TestEncodeWrappedImage` [SKIP] | image/gif, 18m0s |
| [4910231305](https://github.com/golang/go/issues/65171#issuecomment-4910231305) | 2026-07-07 | ppc64le_power9 | `database/sql.TestTypedString/basic` [SKIP] | database/sql, 20m0s |
